src/main/c/NativeLongFunc.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// src/main/c/NativeFunc.cpp - native funcs for Scala NativeLongFunc
+//
+// Richard Cownie, Hail Team, 2018-04-18
+//
+#include "hail/NativePtr.h"
+#include "hail/CommonDefs.h"
+#include <assert.h>
+#include <jni.h>
+
+NAMESPACE_BEGIN(hail)
+
+NAMESPACE_BEGIN_ANON
+
+typedef NativeFuncObj<long> NativeFunc;
+
+NativeFunc* toNativeFunc(JNIEnv* env, jobject thisObj) {
```

Comment:
As per our discussion on variable and function names and the style guide, this should be `to_native_func`, etc.

Reply:
Hmm, that seems like a grey area since "NativeFunc" is the approved form of the type name,
so "to_NativeFunc" might make sense. IDK, that's one place where the different rules for type names and func/method names start to clash.  Guess I'll see if Google examples give any clarity.

Reply:
Fair enough, I don't have a strong opinion on this one.